### PR TITLE
Add fetchMyTrade, fetchDeposits, and fetchWithdrawals to coinbase

### PR DESCRIPTION
**Pull request by milesalex**
WIP. More for discussion...

### CODE REVIEW

**Comment by kroitor** [comment]
Hi! 

Thanks for your involvement!

First of, to understand the inner workings of it, please, read the following:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

We can't really use pure JS, therefore we use a "C-flavor" of it – no maps, closures or comprehensions.  This is the reason why this build is broken. The reason behind it is explained in the docs above. Also, in order to hint the transpiler, we have to keep code formatting very clean and precise to a character. Yes, we know, this may seem a bit harder than just coding in pure JS at first, but, ultimately, this is how we unify JS/Py/PHP in our project and the pros for this approach overweigh the cons.

**Comment by milesalex** [comment]
@kroitor Woah! Thank you for all the help. This is great for me to see your process and how to think about implementing api calls.

I can try to implement deposits and withdrawals using this methodology and get feedback from you?

**Comment by milesalex** [comment]
@kroitor Hmm, seems like the `coinbasepro` test failed. Not sure what I touched to affect that test.

**Comment by kroitor** [comment]
> @kroitor Hmm, seems like the coinbasepro test failed. Not sure what I touched to affect that test.

It's not related to your edits, no worries ) Coinbase Pro time is drifting, breaking one of the tests occasionally. We will either fix the test or the exchange implementation or both, i've silenced it for now within this branch, as you can see from my commits.

**Comment by milesalex** [comment]
Awesome, let me know if you'd like any changes

File: js/coinbase.js [hunk 1]
```diff
@@ -158,6 +158,261 @@ module.exports = class coinbase extends Exchange {
...
+            }
+        }
+        let feeCost = this.safeFloat (feeObject, 'amount');
+        let feeCurrencyId = this.safeString (feeObject, 'currency');
+        let feeCurrency = this.commonCurrencyCode (feeCurrencyId);
+        let fee = {
+            'cost': feeCost,
+            'currency': feeCurrency,
+        };
+        return {
```
Comment (kroitor) [inline js/coinbase.js]: The transaction structure has to conform to the following spec: https://github.com/ccxt/ccxt/wiki/Manual#transaction-structure . It is very similar to a trade, but not quite the same. For transactions, we don't have a `price` and `cost`, and we have some other fields like `address`, `tag` and so on. Will have to rework it a bit, your help is appreciated )

Comment (milesalex) [reply js/coinbase.js]: You got it. also i think i need to verify the correct timestamps. WIll follow up soon.

**Comment by milesalex** [comment]
Woo!! Thank you for the help!

**Comment by kroitor** [comment]
@milesalex thx to you) let us know if anything!

**Pull request merged**